topics/pubsub.md

```diff
@@ -169,6 +169,19 @@ patterns the client is still subscribed to. So the client will exit
 the Pub/Sub state only when this count drops to zero as a result of
 unsubscribing from all the channels and patterns.
 
+## Sharded pubsub
+
+From 7.0, sharded pubsub is introduced in which shard channels are assigned to slots by the same algorithm used to assign keys to slots. 
+A shard message must be sent to a node that own the slot the shard channel is hashed to. 
+The cluster makes sure the published shard messages are forwarded to all the node in the shard, so clients can subscribe to a shard channel by connecting to any one of the nodes in the shard.
```

Comment:
We have this sentence repeated in three places, but I think it might be confusing. I assume that `all the node in the shard` is just a typo, and what was meant was `all the nodes in the cluster`, but if so,  it's not clear to me why do we have sharded pubsub if the published shard messages are still forwarded to all nodes in the cluster, just like regular pub sub. 
I tested this in a cluster and when I sent a message with SPUBLISH on Master 1, I didn't receive it on Master 2, which is what this sentence implies, as far as I understand.

Reply:
It's not a typo. "All the nodes in the shard" means the master that owns a cluster slot, and its replicas (and possibly sub-replicas).

But I understand this can be confusing. It is unusual to refer to "all nodes in the same shard". I think it should instead be described in terms of the master and its replicas.

Suggestion: "The cluster makes sure the published shard messages are forwarded to the replicas, so clients can subscribe to a shard channel by connecting either to the master responsible for the slot, or to any one of its replicas."

@hpatro WDYT?

Reply:
@zuiderkwast I honestly think we should be pushing the notion of a shard more generally within Redis. I dislike having to say "primaries and replicas".

Also, cluster doesn't support replica chaining, so it's not actually a problem for sub-replicas.

Reply:
@madolson I'm all for promoting this notion, but it needs to be thoroughly explained in at least one place on the cluster intro page (preferably with ascii drawings).

So far, I've mainly though of a shard as a set of slots, rather than a set of nodes.

Reply:
Revised suggestion: "The cluster makes sure the published shard messages are forwarded to replicas, so clients can subscribe to a shard channel by connecting any one of the nodes in the shard."

Reply:
Thank you for explaining @zuiderkwast, this makes everything right again :) Yes, to me it was indeed unusual to see a master-replicas set referred to as a shard, just because I had never seen that definition before; personally I prefer the (admittedly) more verbose, but more clear option in your first suggestion:
"The cluster makes sure the published shard messages are forwarded to its replicas, so clients can subscribe to a shard channel by connecting either to the master responsible for the slot, or to any one of its replicas."

Reply:
That wording isn't technically correct though. It's not forwarded to its replicas, its forwarded to *ALL* nodes in the shard, which includes the primaries if it's sent to the replica. The correct phrasing would be:
```
"The cluster makes sure the published shard messages are forwarded to its replicas or master, so clients can subscribe to a shard channel by connecting either to the master responsible for the slot, or to any one of its replicas."
```
perhaps I can be weird and suggest a slight variant:
```
"The cluster makes sure the published shard messages are forwarded to all nodes in the shard, so clients can subscribe to a shard channel by connecting to either the master responsible for the slot, or to any of its replicas"
```

Reply:
@madolson @elena-kolevska @zuiderkwast I'm proceeding with Madelyn's second suggestion. It would let us propagate the "nodes in the shard" concept as well as make it's clear on the usage.

Reply:
@madolson ah, of course! Cause you can publish from a replica. I missed that angle.

Reply:
LGTM